src/main/java/com/mathworks/ci/TestResultsViewAction.java

```diff
@@ -0,0 +1,265 @@
+
+        // Check if test's file was known or not
+        MatlabTestFile matlabTestFile = map.get(baseFolder + File.separator + matlabTestFileName);
+        if(matlabTestFile == null) {
+            matlabTestFile = new MatlabTestFile();
+            matlabTestFile.setName(matlabTestFileName);
+
+            map.put(baseFolder + File.separator + matlabTestFileName, matlabTestFile);
+            testSessionResults.add(matlabTestFile);
+        }
+
+        // Calculate the relative path
+        Path path1 = Paths.get(baseFolder.toURI());
+        Path path2 = Paths.get(this.workspace.toURI());
+        Path relPath = path2.relativize(path1);
+
+        matlabTestFile.setPath(this.workspace.getName() + File.separator + relPath.toString());
+
+        MatlabTestCase matlabTestCase = new MatlabTestCase();
+        matlabTestCase.setName(matlabTestCaseName);
+        matlabTestCase.setPassed((boolean) matlabTestCaseResult.get("Passed"));
+        matlabTestCase.setFailed((boolean) matlabTestCaseResult.get("Failed"));
+        matlabTestCase.setIncomplete((boolean) matlabTestCaseResult.get("Incomplete"));
+        if (matlabTestCaseResult.get("Duration") instanceof Long) {
+            matlabTestCase.setDuration(((Long) matlabTestCaseResult.get("Duration")).doubleValue());
+        } else if (matlabTestCaseResult.get("Duration") instanceof Double) {
+            matlabTestCase.setDuration(((Double) matlabTestCaseResult.get("Duration")));
+        }
+        matlabTestCase.updateStatus();
+
+        Object diagnostics = ((JSONObject)matlabTestCaseResult.get("Details")).get("DiagnosticRecord");
+        if(diagnostics instanceof JSONObject) {
+            MatlabTestDiagnostics matlabTestDiagnostics = new MatlabTestDiagnostics();
+            matlabTestDiagnostics.setEvent(((JSONObject)diagnostics).get("Event").toString());
+            matlabTestDiagnostics.setReport(((JSONObject)diagnostics).get("Report").toString());
+            matlabTestCase.getDiagnostics().add(matlabTestDiagnostics);
+        }
+        else if(diagnostics instanceof JSONArray && ((JSONArray)diagnostics).size() > 0) {
+            Iterator<JSONObject> diagnosticsIterator = ((JSONArray)diagnostics).iterator();
+            while(diagnosticsIterator.hasNext()) {
+                JSONObject diagnosticItem = diagnosticsIterator.next();
+
+                MatlabTestDiagnostics matlabTestDiagnostics = new MatlabTestDiagnostics();
+                matlabTestDiagnostics.setEvent(diagnosticItem.get("Event").toString());
+                matlabTestDiagnostics.setReport(diagnosticItem.get("Report").toString());
+                matlabTestCase.getDiagnostics().add(matlabTestDiagnostics);
+            }
+        }
+
+        matlabTestFile.incrementDuration(matlabTestCase.getDuration());
+        matlabTestFile.updateStatus(matlabTestCase);
+        matlabTestFile.getMatlabTestCases().add(matlabTestCase);
+        updateCount(matlabTestCase);

```

Comment:
Maybe we can have an `addTestCase` method on the MatlabTestFile that handles these responsibilities.

Reply:
Done, thanks!